lib/properties.js

```diff
@@ -129,11 +132,14 @@ class Properties extends EventEmitter {
       // Merge layers' properties into the global property object
       Properties.merge(properties, persistent);
 
-      this.persistent = persistent;
-      this.properties = properties;
-      this.emit('build', this.properties);
+      this.tokendTransformer.transform(properties)
+          .then((transformedProperties) => {
+            this.persistent = persistent;
+            this.properties = Immutable.Map(properties).mergeDeep(transformedProperties).toJS();
```

Comment:
This is intentional. Our `Properties.merge` function won't overwrite existing values with `null` values. We're using `null` to indicate that a secret wasn't found. `Immutable.mergeDeep` will overwrite the existing `$tokend` objects in the properties with either resolved secrets or `null` values.